### DISCUSSION TITLE
New ideas and suggestions I got after some use

### DISCUSSION THREAD

**Discussion by Vombato** [StudyLions/StudyLion#23] [Ideas]
Hi!
My name is **Luca**, I'm an Italian CS & Engineering student, in my free-time I manage/administrate a study community mainly in support of students preparing for admission tests (med, engineering, nursery, vet etc...) and for a while we've been looking for an efficient Discord Bot for our study rooms.
I am really happy that last week I found about StudyLion, since I added it to our server it has been great! It mainly does everything other bots already do, but all in one bot and pretty much better in every way. You really did a ton of work on this!

That being said, I'm here to offer some ideas and concerns about the bot that came to my mind after tinkering a bit with it :smile: 

- **Use of discord.py API and slash commands:** this API wrapper has been discontinued since last year and will not receive updates in the future, also the use of slash commands will be required and enforced for bots used in more than 75 servers by the end of April (if I'm not mistaking). The "simplest" solution would be to migrate to [nextCord](https://github.com/nextcord/nextcord/) or [pyCord](https://github.com/Pycord-Development/pycord) which are just two forks with more features implemented into it (like slash commands, indeed), migrating should be pretty much straight forward as they are almost fully backward compatible.

- **Localization:** I thought that a very nice feature would be implementing a localization feature, like a command to set the desired language for the bot. This could be a pretty heavy reformat but I think could be really useful, especially for asian communities, spanish ones (and also for other languages including Italian :grin:). After setting up the PO Localization many users could help with the translation via platforms like crowdlin or just by PRs (me included).

I give my full availability to help from the end of the month onwards (I need the exams session to end :rofl:)

**Comment by ShootingKing-AM** [comment]
Hello !
Localization is a really good idea to esp to further increase the StudyLion community to non-english speaking crowd. Really nice suggestion, should atleast implement top10 most spoken languages (there is a very large chunk of people who dont really understand English or wish to UI/UX in english)

Btw you can follow the Project's Kanban board on Trello [here](https://trello.com/b/0zOxkqyO/studylion-bot-dev). The SDK migration is already a planned task IIRC

**Reply by Vombato** [reply]
That's great! I just gave it a quick look, you really are working on a ton of stuff, that's remarkable.
Thanks for taking the time to reply to my suggestions, I really just opened the repo, saw an almost empty discussion tab and thought of writing down something! :rofl: 
Jokes apart, keep up the great work and don't hesitate to let me know if I can be of any help! :smiley:

**Comment by AriHoresh** [comment]
Hey Luca! 
Thank you so much for the suggestions. 
I'm glad you enjoy this project. We do plan to migrate to slash commands and, in the future, offer some more languages! 
If you would like to contribute to the code, before you do, make sure to contact me at contact@arihoresh.com so we can discuss it further!
Ari.
